src/suspense.js

```diff
@@ -7,33 +7,37 @@ import { toChildArray } from './diff/children';
 export function Suspense(props) {
 	this.shouldRenderFallback = this.shouldRenderFallback.bind(this);
 	this.shouldRenderTextNode = this.shouldRenderTextNode.bind(this);
-	this.suspendedDom = [];
 	this.$$children = toChildArray(props.children).map(child => {
 		if (typeof child === 'string') {
 			return <SuspenseTextChild shouldRender={this.shouldRenderTextNode}>{child}</SuspenseTextChild>;
 		}
 
 		return child;
 	});
+	this.pendingSuspensions = [];
 }
 Suspense.prototype = new Component();
 
 /**
  * @param {Promise} promise The thrown promise
  */
 Suspense.prototype._childDidSuspend = function(promise, suspendingComponent) {
+	this.pendingSuspensions.push(promise);
```

Comment:
Instead of using an Array here, we can stack promises:

```js
Suspense.prototype._childDidSuspend = function(promise, suspendingComponent) {
  this._pending = this._pending.then(() => promise);
};
```

This means awaiting the pending value is as simple as `this._pending.then`.

Reply:
That's a good point. One problem is that I'd need to unregister the `then` handler of `this._pending` before re-assigning it. Otherwise the suspension would be completed when the first promise throws:

- catch first promise: `this._pending = firstPromise`
  - `this._pending.then(suspensionIsDone)` (de facto `firstPromise.then(suspensionIsDone)`)
- catch. first promise: `this._pending = this.pending.then(() => secondPromise))`
  - need to unregister `firstPromise.then(suspensionIsDone)` and replace with:
  - `this._pending.then(suspensionIsDone)`